src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (!herb.hidden && herb instanceof Herbivore) {
+      herb.health -= 50;
+      Animal.alive = Animal.alive.filter((el) => el.health !== 0);
```

Comment:
The check for `herb.hidden` and `herb instanceof Herbivore` is correct, but after decreasing the health, you should also check if the health is less than or equal to 0 to remove the animal from the alive array. Right now, the code only removes animals with exactly 0 health, which might not catch animals with negative health values.